api/src/test/java/org/confeti/controllers/core/ReportControllerTest.java

```diff
@@ -144,17 +123,151 @@ public void testGetReportTagRequest() {
+                        Report.builder("title3")
+                                .language("EN")
+                                .build(),
+                        Report.builder("title4")
+                                .language("RU")
+                                .build(),
+                        Report.builder("title5")
+                                .language("FR")
+                                .build()
+                )));
+        testGetReportInfoRequestWithConferenceName(
+                String.format("/api/rest/report/language?year=%d&conference_name=%s", year, conferenceName),
+                response);
+    }
+
+    @Test
+    public void testGetReportLanguageRequestWithConferenceName() {
+        final Conference conference1 = Conference.builder("test", 1971).build();
+        final Conference conference2 = Conference.builder("test", 1972).build();
+
+        final ReportResponse response = new ReportResponse(conference1.getName(), Map.of(1971, Map.of("RU", 2L),
+                1972, Map.of("EN", 1L, "RU", 1L)));
+
+        when(reportService.findBy(conference1.getName()))
+                .thenReturn(Flux.fromIterable(Arrays.asList(
+                        Report.builder("title")
+                                .language("RU")
+                                .conferences(Set.of(conference1))
+                                .build(),
+                        Report.builder("title11")
+                                .language("RU")
+                                .conferences(Set.of(conference1))
+                                .build(),
+                        Report.builder("title2")
+                                .language("EN")
+                                .conferences(Set.of(conference2))
+                                .build(),
+                        Report.builder("title22")
+                                .language("RU")
+                                .conferences(Set.of(conference2))
+                                .build()
+                )));
+        testGetReportInfoRequestWithConferenceName(String.format("/api/rest/report/language?conference_name=%s", conference1.getName()), response);
+    }
+
+    @Test
+    public void testGetReportLanguageRequestWithYear() {
+        final int conferenceYear = 1971;
+        final Conference conference1 = Conference.builder("test", conferenceYear).build();
+        final Conference conference2 = Conference.builder("test", 1972).build();
+        final Conference conference3 = Conference.builder("test2", conferenceYear).build();
+
+        when(reportService.findAll())
+                .thenReturn(Flux.fromIterable(Arrays.asList(
+                        Report.builder("title")
+                                .language("RU")
+                                .conferences(Set.of(conference1))
+                                .build(),
+                        Report.builder("title11")
+                                .language("Ru-ru")
```

Comment:
`Ru-ru`? Now we save the language as an ISO 639 alpha-2 or alpha-3 language code [see](https://github.com/confeti/confeti/blob/develop/api/src/main/java/org/confeti/db/model/report/AbstractReportEntity.java#L39). But we can save language with country code like `Ru-ru` and I think we have to decide in what form we will save it.

Reply:
It's just a test.
I assume you convert provided language string if needed